experimental/credentials/tls.go

```diff
@@ -0,0 +1,227 @@
+/*
+ *
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package credentials contains experimental TLS credentials.
```

Comment:
This should have a big scary comment about how these credentials should only be used to maintain compatibility for people relying upon interoperability with clients violating the HTTP/2 specification, and that it will be deleted, so users should not use it directly.  Instead they either need to vendor gRPC or copy these credentials.

Reply:
Updated the package comment, please take a look.